File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    
    int zero = 0;
    int cnt = 0;
    
    while(s != "1"){
        string tmp = "";
        cnt++;
        
        for(int i = 0; i<s.size(); i++){
            if(s[i] == '0'){
                zero++;
            }
            else{
                tmp += '1';
            }
        }
        
        s = "";
        for(int i = tmp.size(); i>0; i/=2){
            s+=to_string(i%2);
        }
    }
    answer.push_back(cnt);
    answer.push_back(zero);
    return answer;
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        
        while(!s.equals("1")) {
            for(char c : s.toCharArray()){
                if(c == '0'){
                    answer[1]++;
                }
            }
            s = s.replace("0", "");
            
            int x = s.length();
            StringBuilder sb = new StringBuilder();
            while(x != 0) {
                sb.append(x % 2);
                x /= 2;
            }
            s = sb.reverse().toString();
            answer[0]++;
        }
        
        return answer;
    }
}